src/nodes/hypertable_modify.c

```diff
@@ -745,7 +1027,75 @@ ExecModifyTable(PlanState *pstate)
 	context.mtstate = node;
 	context.epqstate = &node->mt_epqstate;
 	context.estate = estate;
+	/*
+	 * For UPDATE/DELETE on compressed hypertable, decompress chunks and
+	 * move rows to uncompressed chunks.
+	 */
+	if ((operation == CMD_DELETE || operation == CMD_UPDATE) && !ht_state->comp_chunks_processed)
+	{
+		collect_chunks_from_scan(pstate, ht_state);
+		ListCell *ch;
+		int pos = 0;
+		foreach (ch, ht_state->chunks)
+		{
+			bool chunk_status_changed = false;
+			PredicatesContext *cx = NULL;
+			List *scankeys = NULL;
+			List *null_columns = NULL;
+			List *is_null_checks = NULL;
```

Comment:
```suggestion
			List *scankeys = NIL;
			List *null_columns = NIL;
			List *is_null_checks = NIL;
```
The correct macro for empty list is NIL

Reply:
Fixed.